sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -5407,5 +5640,10 @@ public void merge(RequestOptions requestOptions) {
                 }
             }
         }
+
+        public void reset() {
+            this.createdDiagnostics.clear();
```

Comment:
Trying to understand the design here:

- For every response (mapped to a request) a merge of the `CosmosDiagnostics` instance is done (could be cancellation or an error response as well).
- A `reset` is done when a given "operation". For query - 1 "operation" is when 1 `FeedResponse` is obtained. Preceding a `reset` is when you create a new `CosmosDiagnosticContext` instance with a new sequence no.

Reply:
Correct - The DiagnosticsFactory will capture all CosmosDiagnostics instances - when reset is called it clears the list of captured "CosmosDianostics instances created" - merge will make sure all currently captured diagnostics instances will be mapped to the current CosmosDiagnosticsContext

The overall goal for PagedFlux is
Ever FeedResponse bubbled up to the application will have its own ComosDiagnosticsContext instance - which will have potentially multipel CosmosDiagnostics instances (for example 1 for query plan, 1 for request to first region, one for hedign request to second region). With aggregates One FeedResponse bubbled up to application could also mean we had to reach out to multiple partitions internally (but those would be different request statistics within one ComosDiagnostics)

To achieve this we have the diagnostics factory capturing all diagnostics being created - whenever we bubble up a new feedresponse we merge (map the current diagnostics context to all previously captured diagnostics) and then clear the list of captured contexts)